src/aiod_sdk/calls/calls.py

```diff
@@ -0,0 +1,165 @@
+
+
+def get_asset(
+    asset_type: str,
+    identifier: int,
+    version: str | None = None,
+    data_format: Literal["pandas", "dict"] = "pandas",
+) -> pd.Series | dict:
+    """
+    Retrieve metadata for a specific ASSET_TYPE.
+
+    Parameters:
+        identifier (int): The identifier of the ASSET_TYPE to retrieve.
+        version (str | None): The version of the endpoint (default is None).
+        format (Literal["pandas", "dict"]): The desired format for the response (default is "pandas").
+
+    Returns:
+        pd.Series | dict: The retrieved metadata for the specified ASSET_TYPE.
+    """
+    url = url_to_get_asset(asset_type, identifier, version)
+    res = requests.get(url)
+    resources = format_response(res.json(), data_format)
+    return resources
+
+
+async def get_asset_async(
+    asset_type: str,
+    identifiers: list[int],
+    version: str | None = None,
+    data_format: Literal["pandas", "dict"] = "pandas",
+):
+    """
+    Asynchronously retrieve metadata for a list of ASSET_TYPE identifiers.
+
+    Parameters:
+        identifiers (list[int]): The list of identifiers of the ASSET_TYPE to retrieve.
+        version (str | None): The version of the endpoint (default is None).
+        format (Literal["pandas", "dict"]): The desired format for the response (default is "pandas").
+
+    Returns:
+        pd.Series | dict: The retrieved metadata for the specified ASSET_TYPE.
+    """
+    urls = [
+        url_to_get_asset(asset_type, identifier, version) for identifier in identifiers
+    ]
+    response_data = await _fetch_resources(urls)
+    resources = format_response(response_data, data_format)
+    return resources
+
+
+async def get_list_async(
+    asset_type: str,
+    offset: int = 0,
+    limit: int = 100,
+    batch_size: int = 10,
+    version: str | None = None,
+    data_format: Literal["pandas", "dict"] = "pandas",
+) -> pd.DataFrame | dict:
+    """
+    Asynchronously retrieve a list of ASSET_TYPE from the catalogue in batches.
+
+    Parameters:
+        offset (int): The offset for pagination (default is 0).
+        limit (int): The maximum number of items to retrieve (default is 10).
+        batch_size (int): The number of items in a a batch.
+        version (str | None): The version of the endpoint (default is None).
+        data_format (Literal["pandas", "dict"]): The desired format for the response (default is "pandas").
+
+    Returns:
+        pd.DataFrame | dict: The retrieved metadata in the specified format.
+    """
+    if batch_size <= 0:
+        raise ValueError(
+            "batch_size must be larger than 0, otherwise you can use the synchronous get_list function!"
+        )
+    current_offset = offset
+    next_limit = min(batch_size, limit)
+    batches = []
+    while current_offset < offset + limit:
+        batches.append((current_offset, next_limit))
+        current_offset += next_limit
+        next_limit = min(batch_size, offset + limit - current_offset)
```

Comment:
```suggestion
    offsets = range(offset, offset + limit, batch_size)
    # if the limit is not an exact multiple of batch_size, need to truncate the last call
    last_batch_size = (limit % batch_size) or batch_size  
    batch_sizes = [batch_size] * (len(offsets) - 1) + [last_batch_size]
```
and then replace `batches` with `zip(offsets, batch_sizes)` below